src/assets/a11y-viewer.js

```diff
@@ -1,169 +1,136 @@
+        
+        collapsibleHandlers.forEach(function (collapsible) {
+            collapsible.addEventListener('click', function() {
+                collapsibleTarget.classList.toggle(OPEN_CLASS);
+            }, false);
+        });
     }
 
-    function restoreForm(state) {
-
-        formElements.forEach(function (element) {
-            var newValue = state[element.id];
+    function bindEvents() {
+        FORM.addEventListener('change', function(event) {
+            updateView(event.target)
+        }, false);
 
-            if (element.type == "checkbox" && element.checked.toString() != newValue) {
-                element.checked = newValue == "true";
-                update(element, false);
-            } else if (element.value != newValue) {
-                element.value = newValue;
-                update(element, false);
-            }
+        // the `change` event is fired as well when user submits form,
+        // so we don't need to do anything here
+        FORM.addEventListener('submit', function(event) {
+            event.preventDefault();
+        }, false);
+
+        // When the back/forward button is used update the form
+        window.addEventListener('popstate', function() {
+            populateForm();
         });
     }
 
-    function update(formElement, isTriggeredByUser) {
-        var value;
-
-        switch (formElement) {
-            case visionSelect:
-                value = formElement.value;
-                simulateVisionLoss(value);
-                break;
-            case colorSelect:
-                value = formElement.value;
-                simulateColorBlindness(value);
-                break;
-            case keyboardOnlyInput:
-                value = formElement.checked;
-                toggleClickStealer(value);
-                break;
-            case urlInput:
-                value = formElement.value;
-                updateTestSite(value);
-        }
-
-        if (supportsPushState && isTriggeredByUser) {
-            updateHistory(formElement.id, value);
-        }
+    // Update the URL and the view
+    function updateView(formElement) {
+        var actions = {
+            'vision': simulateOption,
+            'color': simulateOption,
+            'keyboard-only': toggleClickStealer,
+            'url': updateTestSite
+        };
+
+        actions[formElement.name](formElement);
+        updateUrl();
     }
 
-    function simulateVisionLoss(value) {
-
-        visionSelectOptions.forEach(function (option) {
-            main.classList.remove(option.value);
-        });
-
-        main.classList.add(value);
+    // Update the address bar with input/values excluding the submit button
+    function updateUrl() {
+        // Only update the URL when the history API is supported
+        if ('pushState' in window.history) {
+            // Exclude the submit button
+            var inputs = arrayFrom(FORM.elements).filter(function (element) {
```

Comment:
shall we remove the arrayfrom function and just use `Array.prototype.filter.call` instead? or map, reduce, foreach, etc